server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/resources/cucumber/GetMailboxes.feature

```diff
@@ -0,0 +1,154 @@
+    Then a mailboxes answer is returned without error
+    And the list should contain 7 mailboxes
+    And the mailbox "shared" should be present
+    And the mailbox "shared" as 1 messages
+    And the mailbox "shared" as 1 unseen messages
+
+  Scenario: Move message should update the total and the unread counts when asked by sharee
+    Given "bob@domain.tld" has a message "m1" in "bobMailbox" mailbox with subject "my test subject", content "testmail"
+    And "bob@domain.tld" moves "m1" to mailbox "shared" of user "alice@domain.tld"
+    When "alice@domain.tld" ask for mailboxes
+    Then a mailboxes answer is returned without error
+    And the list should contain 6 mailboxes
+    And the mailbox "shared" should be present
+    And the mailbox "shared" as 1 messages
+    And the mailbox "shared" as 1 unseen messages
+
+  Scenario: Move message should update the total and the unread counts when asked by sharee / sharee view
+    Given "bob@domain.tld" has a message "m1" in "bobMailbox" mailbox with subject "my test subject", content "testmail"
+    And "bob@domain.tld" moves "m1" to mailbox "shared" of user "alice@domain.tld"
+    When "bob@domain.tld" ask for mailboxes
+    Then a mailboxes answer is returned without error
+    And the list should contain 7 mailboxes
+    And the mailbox "shared" should be present
+    And the mailbox "shared" as 1 messages
+    And the mailbox "shared" as 1 unseen messages
+
+  Scenario: Moving a message to a delegated mailbox without rights should not change the total and the unread counts
+    Given "alice@domain.tld" has a mailbox "shared2"
+    And "alice@domain.tld" shares her mailbox "shared2" with "bob@domain.tld" with "lr" rights
+    And "bob@domain.tld" has a message "m1" in "bobMailbox" mailbox with subject "my test subject", content "testmail"
+    And "bob@domain.tld" moves "m1" to mailbox "shared2" of user "alice@domain.tld"
+    When "alice@domain.tld" ask for mailboxes
+    Then a mailboxes answer is returned without error
+    And the list should contain 7 mailboxes
+    And the mailbox "shared" should be present
+    And the mailbox "shared" as 0 messages
+    And the mailbox "shared" as 0 unseen messages
+
+  Scenario: Moving a message to a delegated mailbox without rights should not change the total and the unread counts / sharee view
+    Given "alice@domain.tld" has a mailbox "shared2"
+    And "alice@domain.tld" shares her mailbox "shared2" with "bob@domain.tld" with "lr" rights
+    And "bob@domain.tld" has a message "m1" in "bobMailbox" mailbox with subject "my test subject", content "testmail"
+    And "bob@domain.tld" moves "m1" to mailbox "shared2" of user "alice@domain.tld"
+    When "bob@domain.tld" ask for mailboxes
+    Then a mailboxes answer is returned without error
+    And the list should contain 8 mailboxes
+    And the mailbox "shared" should be present
+    And the mailbox "shared" as 0 messages
+    And the mailbox "shared" as 0 unseen messages
```

Comment:
I think there is some missing test. When the message is shared, what happens when it is mark as read by the share or by sharer, from the point of you of the sharee, and of the sharer.

Reply:
I was about to ask the same thing (I was actually checking the ticket description but it's missing from it too)

Reply:
We also miss:

 - Copying a Seen flag with lri rights should update total count, but not unseen count.
 - Does "l" right allow me to see "total" and "unseen"